### PR TITLE
[Codegen] Fix if_then_else codegen

### PR DESCRIPTION
**Pull request by spectrometerHBH**
- Evaluating `op.if_then_else` with `a?b:c` is potentially dangerous as it might evaluate expressions in branches without a guard of condition.

- While loop condition expression might require several var bindings generated, so we correct the while loop codegen from `while (cond)` to `while (True)`

### CODE REVIEW

File: src/runtime/ndarray.cc [hunk 1]
```diff
@@ -96,6 +96,27 @@ void ArrayCopyToBytes(const DLTensor* handle, void* data, size_t nbytes) {
   DeviceAPI::Get(handle->device)->StreamSync(handle->device, nullptr);
 }
 
+TVM_REGISTER_GLOBAL("mlc.show_dltensor").set_body_typed([](NDArray arr) {
```
Comment (junrushao) [inline src/runtime/ndarray.cc:99]: remove debug code

**Comment by vinx13** [comment]
isn't short circuiting the defined behavior for ternary expression?

**Comment by spectrometerHBH** [comment]
> isn't short circuiting the defined behavior for ternary expression?

When generating codes for `a?b:c`, there might be bindings for b or c generated ahead. These binding codes are not protected by condition a.

**Review by jinhongyii — APPROVED** [review]

**Comment by ysh329** [comment]
> > isn't short circuiting the defined behavior for ternary expression?
> 
> When generating codes for `a?b:c`, there might be bindings for b or c generated ahead. These binding codes are not protected by condition a.

Hi, bohan, I recently am supporting Ternary Operator(like cond?true_branch:false_branch) in our device. I don't understand this modification. Can u explain this with a case? Thanks a lot!

**Pull request merged**